Add GenAI apis that are also used for translation

tovmasharrison opened the pull request:
# Description

Added 7 GenAI apis that are also used for translation. Additionally, I needed to do a quick syntax fix on `generate.py`, and add some languages to `languages.json`.

### Checklist:

- [x] I have read the [contributing guidelines](/CONTRIBUTING).
- [x] I have followed the [style guide](http://machinetranslate.org/style).

bittlingmayer commented:
Should we add bits to indicate the following?
- distinguish genAI from other APIs
- distinguish translation-specific genAI from other genAI?

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,836 @@
+        "speech_translation": true,
+        "file_translation": true,
+        "self-serve": true,
+        "gen_ai": true
```
there's basically 3 groups:

1. "traditional" APIs
2. GenAI, but translation-specific (TowerLLM, Lara...) i.e. it does not require a prompt
3. GenAI, not translation-specific, so it requires a prompt

so some options:

- `"translation_specific": false`
- `"type": "gen_ai" / "translation_gen_ai" / ...`
- `"prompt_required": true`

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,836 @@
+            "zh-cn",
+            "zh-tw"
+        ],
+        "stock_api_domains": [
```
is this actually true?  do these really exist?

tovmasharrison replied:
It's in their marketed solutions.

tovmasharrison replied:
Has to be removed

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,836 @@
+            "ro",
+            "rof",
+            "rom",
+            "root",
```
is this confirmed?  a 4-letter code seems odd

tovmasharrison replied:
You're right. I have removed it.

tovmasharrison commented:
> * we need to distinguish generic genAI from translation-specific genAI
> * we need to distinguish models from APIs

All done:

I only added two types of flags (i.e. type, and access_method)

- If "type": "translation_gen_ai/gen_ai/translation_gen_ai_and_mt/mt", then by default prompt_required flag and translation_specific flag not required, as it covers all based on the exact type. 

So if:
- "type": "translation_gen_ai", then automatically, "prompt_required": false, and "translation_specific": true
- "type": "gen_ai", then automatically, "prompt_required": true, and "translation_specific": false
- and so on

For **access_method**, it will have two options, **api** or **model**
- "access_method": "api"
- "access_method": "model"

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,842 @@
+            "vi",
+            "zh"
+        ],
+        "text_translation": true,
```
or input_modes and output_modes, if we find those are different.

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,842 @@
         ]
+    },
+    {
+        "name": "OpenAI GPT",
```
I guess we should make it about the *api*, not about a model.

So the API is OpenAI (https://platform.openai.com/docs/overview)

Then there are multiple models that can do translation (GPT-*, Whisper-*), not all endpoints support all models, it gets complicated fast.

e.g. this supports speech-to-text translation 
https://platform.openai.com/docs/guides/speech-to-text/translations#supported-languages

And they also offer an text-to-speech API, so effectively you can get speech-to-speech translation.

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,842 @@
+        "file_translation": true,
+        "self-serve": true,
+        "type": "gen_ai",
+        "access_method": "model"
```
I don't love this, because:
- The traditional APIs also all have underlying models.  They're just less transparent about it.
- All of them are accessed via an API.  It's just that these ones require additional prompting (to be passed to the API)
- We don't want to be forced to go add `"access_method": "api"` to all of the APIs we already covered.

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,836 @@
+        "gen_ai": true
+    },
+    {
+        "name": "TowerLLM",
```
Sounds good.

That will also help us distinguish between GPT and Whisper, within the OpenAI API.

But then in that case, should we have a models.json?

tovmasharrison commented:
- Added a separate `models.json` file, where I added the models, which are used by the GenAI APIs. 
- Created a separate Models section with it's **coming soon** page, and a page for each model. 
-  If a model is open, I added the link to the downloadable model, which is mainly through Hugging Face.
-  Added **models** flag for APIs to indicate what model(s) they are using, and displayed them on each API page, if any.
- Removed **TowerLLM** and **Meta NLLB** from the `apis.json` file, and added them to `models.json`, as they're models and they shouldn't be under APIs.
- Added **Widn** api
- Changed how inputs types for the GenAI APIs are given. Now they are added as list items. I will also modify this in the other [PR](https://github.com/machinetranslate/machinetranslate.org/pull/641).

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "self_serve": true
+    },
+    {
+        "name": "Sarvam",
```
seems like its name is Sarvam-Translate

so maybe we should make the name and id reflect that?

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "company": "meta",
+        "urls": [
+            "https://huggingface.co/facebook/nllb-200-distilled-600M",
+            "https://ai.facebook.com/research/no-language-left-behin"
```
typo

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "self_serve": true
+    },
+    {
+        "name": "Meta NLLB",
```
I think without "Meta", just "NLLB"

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "id": "meta-nllb",
+        "company": "meta",
+        "urls": [
+            "https://huggingface.co/facebook/nllb-200-distilled-600M",
```
feels like we should put the URLs of all the model versions, not just one

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "id": "whisper",
+        "company": "openai",
+        "urls": [
+            "https://huggingface.co/openai/models",
```
maybe we can remove this?

tovmasharrison replied:
Ye, better to remove it.

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "company": "openai",
+        "urls": [
+            "https://huggingface.co/openai/models",
+            "https://huggingface.co/openai/whisper-large-v3",
```
ditto, urls for all the models?

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+        "urls": [
+            "https://huggingface.co/openai/models",
+            "https://huggingface.co/openai/whisper-large-v3",
+            "https://github.com/openai/whisper"
```
https://platform.openai.com/docs/models/whisper-1 too

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,122 @@
+       "id": "voxtral",
+       "company": "mistral-ai",
+       "urls": [
+           "https://huggingface.co/mistralai/Voxtral-Mini-3B-2507",
```
all the models

bittlingmayer commented on _layouts/model.html:
```diff
@@ -0,0 +1,108 @@
+
+<hr/>
+
+{% if page.active != true %}
```
I guess for a model, it's not possible.

bittlingmayer commented on _layouts/model.html:
```diff
@@ -0,0 +1,108 @@
+  {% endif %}
+</p>
+
+<p>
```
I think we can delete this whole section.  For models it doesn't make sense.

tovmasharrison replied:
Deleted

bittlingmayer commented on _layouts/model.html:
```diff
@@ -0,0 +1,108 @@
+</blockquote>
+
+<section id="integrations">
+  <h2>Integrations</h2>
```
I'd call it "APIs"

And phrase is as "available via", not "integrated in".

"TowerLLM is available via 1 translation API."

bittlingmayer commented on _layouts/model.html:
```diff
@@ -0,0 +1,108 @@
+
+<p>
+  {% if page.apis.size == 1 %}
+  <a href="#integrations">1 translation API</a> has integrated the <strong>{{ page.title }}</strong> large language model.{% endif %}
```
```
<a href="#apis">1 translation API</a> serves the <strong>{{ page.title }}</strong> large language model.
```

bittlingmayer commented on _layouts/model.html:
```diff
@@ -0,0 +1,108 @@
+  <h2>Integrations</h2>
+  <details>
+    <summary>
+      {{ page.title }} is integrated in <strong>{{ page.apis | size }}</strong> translation API{% if page.apis.size != 1 %}s{% endif %}.
```
```
{{ page.title }} is served by <strong>{{ page.apis | size }}</strong> translation API{% if page.apis.size != 1 %}s{% endif %}.
```

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,524 @@
+    },
+    {
+        "name": "Claude AI",
+        "id": "claude",
```
ditto

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,138 @@
+        "self_serve": true
+    },
+    {
+        "name": "Sonnet",
```
I think we should call it "Claude Sonnet" / `claude-sonnet`.

bittlingmayer commented on apis/claude.md:
```diff
@@ -0,0 +1,342 @@
+nav_order: 948
+layout: api
+title: Claude AI
+description: The Claude AI machine translation API
```
I think, if it's not translation-specific i.e. if prompting is required to make it do translation, then we should call it e.g. `The Anthropic API for machine translation`.

tovmasharrison replied:
I will do this in #642 as well.

bittlingmayer commented on apis/claude.md:
```diff
@@ -0,0 +1,342 @@
+- https://www.anthropic.com
+privacy_url: null
+self_serve: true
+customisation: []
```
maybe we should add `prompting` as another customisation type?

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,524 @@
+        "company": "anthropic",
+        "urls": [
+            "https://docs.anthropic.com",
+            "https://www.anthropic.com"
```
https://docs.anthropic.com/en/docs/build-with-claude/multilingual-support

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,525 @@
+        "id": "openai",
+        "company": "openai",
+        "urls": [
+            "https://platform.openai.com/docs/guides/gpt",
```
https://github.com/openai/whisper#available-models-and-languages
https://help.openai.com/en/articles/6742369-how-can-i-use-the-openai-api-with-text-in-different-languages

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,525 @@
+        ]
+    },
+    {
+        "name": "Qwen3‑MT",
```
I think we should just call it Qwen-MT / `qwen-mt`.

That's what they call it.

The model version will change too often.

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,525 @@
+        "adaptive": true,
+        "self-serve": true,
+        "models": [
+            "QWEN"
```
Qwen

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,525 @@
+           "pt",
+           "zh"
+       ],
+       "inputs": [
```
text too, right?

bittlingmayer replied:
prompt required, right?

bittlingmayer commented on _data/models.json:
```diff
@@ -0,0 +1,138 @@
+        "self_serve": true
+    },
+    {
+        "name": "QWEN",
```
Qwen

bittlingmayer commented on _data/apis.json:
```diff
@@ -14866,5 +14866,525 @@
+       ],
+       "self-serve": true,
+       "models": [
+        "Voxtral"
```
Mistral and Mixtral too, right?  (e.g. for text)
